examples/nidcpower/measure-record.py

```diff
@@ -180,11 +180,11 @@ def _on_close(event):
 
             # Updating the precision of the fetched values.
             y_axis_new = []
-            for value in y_axis:
-                if value < VOLTAGE_LEVEL:
-                    y_axis_new.append(math.floor(value * 100) / 100)
+            for y_value in y_axis:
```

Comment:
I don't know if this was the best way to address this. But it was complaining on line 208 that we were trying to assign a string to a float variable because `value` was defined here.

Reply:
Yeah, that's correct. Try this

`try.py`:
```python
def test():
    try:
        while True:
            x = "ok"
            raise IndexError
    except IndexError:
        print(x)
        x = 1
        print(x)


test()
```

Output:
```
(.venv) λ python try.py
ok
1
```

`python` variable scope is not the same as C++/C#/etc and `mypy` is generally trying to follow that or at least being conservative to prevent potential confusion if the typechecker doesn't match the runtime. There's a [`--allow-redefinition`](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-allow-redefinition) option, but I think it actually excludes this case. I'm not sure why that is, I think because the runtime behavior is particularly surprising. And because it is more often done on accident, as opposed to reusing in the same/equal scope, which is semi-idiomatic (if a little sketchy).